### PR TITLE
Topic recognition - s4585711

### PR DESCRIPTION
**Pull request by RocksnStuff**
Student number: s4585711
Name: Matthew Harris
Topic: Vision Transformer to classify ADNI dataset

My implementation can be found in [recognition/4585711_VIT_ADNI](../tree/topic-recognition/recognition/4585711_VIT_ADNI)

In the end, I only got a testing accuracy of 57%, I had hoped to get higher.

I could have been doing something wrong, but I didn't find that any data preprocessing (including data augmentation) actually helped. In fact, it seems to make the accuracy even worse. One of the more novel preprocessing techniques I tried was tracking the position of each brain to centre and crop each image. However, this did not work. In the end, the only preprocessing method that actually helped was normalisation.

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
This is an initial inspection, no action is required at this point

ViT design: Great
Data split: Do you have a test set? Or do you only have a training and validation set?
Model training: Validation accuracy and loss don't look right
Commit log: Great

If you split data into train val and test set, you will see validation ACC will be less than training ACC. Test ACC not over 80% is acceptable.

**Comment by RocksnStuff** [comment]
(just realised I made a typo in the initial comment, meant to say 57% testing accuracy, not training accuracy)

I did have a test set as well as the training and validation sets. I got 84.37% training accuracy, 61.79% for validation, and 57.04% for testing.

**Comment by LinfengLiu98** [comment]
> t as well as the training and va

Okay, then that's all good. Great work.

**Comment by shakes76** [comment]
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate use and implementation (convergence issues) -1
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem (low performance) -1
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage 
Module present
Commenting
No Data leakage (test set lacking) -1
Difficulty: Hard
## Commit Log
Meaningful commit messages could be more informative -1
Progressive commits used
## Documentation
ReadMe acceptable/good
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good
